azurelinuxagent/common/protocol/wire.py

```diff
@@ -995,6 +1075,41 @@ def check_wire_protocol_version(self):
                      "advised by Fabric.").format(PROTOCOL_VERSION)
             raise ProtocolNotFoundError(error)
 
+    def send_request_hostplugin_first(self, direct_func, host_func):
+        # For certain calls we want them to go through the HostGAPlugin first, since it
+        # supports the If-None-Match header that reduces how often we need to parse
+
+        # note that this method requires a lambda that returns two values
+        ret = None
+        etag = None
+        not_modified = False
+        try:
+            ret, etag, not_modified = host_func()
+
+            # Different direct channel functions report failure in different ways: by returning None, False,
```

Comment:
Not relevant to this PR, but we should make all the direct_func and host_func more consistent to either use try-catch or the return value for code cleanliness and consistency